client/src/app/current-users/current-users.component.ts

```diff
@@ -12,40 +13,31 @@ import { UsersObject } from '../../../../api-objects/UsersObject';
 export class CurrentUsersComponent implements OnInit {
   public usersTable: UsersObject[] = [];
   public pageCounter: number = 0;
+  public authorityTypes = authorityTypes;
 
-  constructor(/*private currentusersTableService: GetCurrentUsersService*/) { }
+  constructor(private apiUsersCrudService: ApiUsersCrudService, private notificationService: NotificationService) { }
 
   public ngOnInit() {
     this.getAllUsers();
   }
 
   public getAllUsers(): void {
-    //  this.currentUsersService.getUsers()
-    //  .subscribe(
-    //    (results) => {
-    //      this.usersTable = results.data;
-    //      console.log(this.usersTable);
-    //      console.log('Users loading...');
-
-    //  },(error) => {
-    //    console.log('There was an error' + error);
-    //  });
-
-    this.usersTable = Users;
-    // this.usersTable.map((types) => {
-    //   if(types.authorityType === 0) {} else {}
-    // });
+    this.apiUsersCrudService.getUsers()
+      .subscribe(
+        (results) => {
+          this.usersTable = results.data;
+        });
   }
 
   public prevPage(pageNum: number): void {
-    while (this.pageCounter > 0 && pageNum > 0) {
+    if (this.pageCounter > 0 && pageNum > 0) {
       this.pageCounter -= 1;
       pageNum -= 1;
     }
   }
 
   public nextPage(pageNum: number): void {
-    while ((this.pageCounter + 1) * 10 < this.usersTable.length && pageNum > 0) {
+    if ((this.pageCounter + 1) * 10 < this.usersTable.length && pageNum > 0) {
```

Comment:
It's a tiny change, but I'd like to see 10 be turned into a constant so as to remove magic numbers from the function.